apps/language_server/lib/language_server/providers/formatting.ex

```diff
@@ -103,4 +108,63 @@ defmodule ElixirLS.LanguageServer.Providers.Formatting do
       end
     end)
   end
+
+  ## GenServer Callbacks
+  #####################
+  def start_link(opts) do
+    if supported?() do
+      GenServer.start_link(__MODULE__, :ok, opts |> Keyword.put_new(:name, __MODULE__))
+    else
+      :ignore
+    end
+  end
+
+  def init(_) do
+    _ = :ets.new(__MODULE__, [:set, :protected, :named_table, read_concurrency: true])
+    {:ok, nil}
+  end
+
+  def handle_call({:build_cache, dir}, _, state) do
+    opts_result = SourceFile.formatter_opts(dir)
+    :ets.delete_all_objects(__MODULE__)
+    :ets.insert(__MODULE__, {:formatter_opts, opts_result})
+
+    case opts_result do
+      {:ok, opts} ->
+        IO.puts("[ElixirLS FormattingCache] Formatter building cache")
```

Comment:
I tried using the same function as here https://github.com/elixir-lsp/elixir-ls/blob/master/apps/language_server/lib/language_server/providers/workspace_symbols.ex#L130 but got some sort of process error. Will improve.